flashinfer/__init__.py

```
@@ -266,3 +267,47 @@
+    #   inner try  – safely resolve the commit hash; falls back to "unknown"
+    #                if __git_commit__ is missing or malformed so the log
+    #                line is still emitted rather than silently suppressed.
+    #   outer try  – absorbs every other failure (non-integer LOGLEVEL env
+    #                var, closed/None stdout or stderr, unwritable log file)
+    #                so a logging misconfiguration can never block the import.
+    try:
+        if int(_os.environ.get("FLASHINFER_LOGLEVEL", "0")) < 1:
+            return
+        try:
+            _short = __git_commit__[:8] if __git_commit__ != "unknown" else "unknown"
+        except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)